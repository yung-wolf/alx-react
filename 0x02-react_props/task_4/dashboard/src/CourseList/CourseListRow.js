// a new component with the name CourseListRow. The component accepts three props:
// isHeader (bool, default: false)
// textFirstCell (string, required)
// textSecondCell (string, default: null)

import React from 'react';

function CourseListRow({ isHeader=false, textFirstCell, textSecondCell=null }) {
  return (
    <tr>
      {/* {!isHeader && <td>{textFirstCell}</td>}
      {isHeader && <th>{textFirstCell}</th>}
      {textSecondCell && <td>{textSecondCell}</td>} */}

      {isHeader ? (
        textSecondCell == null ?
        <th id="title" colSpan="2">{textFirstCell}</th> :
        <>
          <th>{textFirstCell}</th>
          <th>{textSecondCell}</th>
        </>
        ) : (
          <>
            <td>{textFirstCell}</td>
            <td>{textSecondCell}</td>
          </>
        )}
    </tr>
  )
}

export default CourseListRow