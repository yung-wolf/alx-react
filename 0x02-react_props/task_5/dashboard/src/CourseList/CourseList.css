table {
  width: 100%;
  border: 1px solid lightgrey;
  border-collapse: collapse;
  font-family: var(--font-family-secondary);
}

th {
  border-bottom: 2px solid lightgrey;
  padding: 5px;
}

td {
  padding: 5px;
}

#title {
  text-align: center;
}