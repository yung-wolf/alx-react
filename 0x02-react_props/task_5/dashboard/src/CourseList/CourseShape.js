import Proptypes from 'prop-types';

// function CourseShape({ id, name, credit }) {
//   return
// }

// Define proptypes
const CourseShape = Proptypes.shape({
  id: Proptypes.number.isRequired,
  name: Proptypes.string.isRequired,
  credit: Proptypes.number.isRequired
})

export default CourseShape