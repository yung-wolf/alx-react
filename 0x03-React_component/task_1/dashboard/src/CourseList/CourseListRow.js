// a new component with the name CourseListRow. The component accepts three props:
// isHeader (bool, default: false)
// textFirstCell (string, required)
// textSecondCell (string / number, default: null)

import React from 'react';

function CourseListRow({ isHeader=false, textFirstCell, textSecondCell=null }) {
  return (
    <tr>
      {isHeader ? (
        textSecondCell == null ?
        <th id="title" colSpan="2">{textFirstCell}</th> :
        <>
          <th>{textFirstCell}</th>
          <th>{textSecondCell}</th>
        </>
        ) : (
          <>
            <td>{textFirstCell}</td>
            <td>{textSecondCell}</td>
          </>
        )}
    </tr>
  )
}

export default CourseListRow