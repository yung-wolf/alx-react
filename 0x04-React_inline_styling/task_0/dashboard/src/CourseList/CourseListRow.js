// a new component with the name CourseListRow. The component accepts three props:
// isHeader (bool, default: false)
// textFirstCell (string, required)
// textSecondCell (string / number, default: null)

import React from 'react';

const rowColor = { backgroundColor: '#f5f5f5ab' };
const headerRowColor = { backgroundColor: '#deb5b545' };

function CourseListRow({ isHeader=false, textFirstCell, textSecondCell=null }) {
  return (
    <tr style={isHeader ? headerRowColor : rowColor}>
      {isHeader ? (
        textSecondCell == null ?
        <th id="title" colSpan="2">{textFirstCell}</th> :
        <>
          <th>{textFirstCell}</th>
          <th>{textSecondCell}</th>
        </>
        ) : (
          <>
            <td>{textFirstCell}</td>
            <td>{textSecondCell}</td>
          </>
        )}
    </tr>
  )
}

export default CourseListRow