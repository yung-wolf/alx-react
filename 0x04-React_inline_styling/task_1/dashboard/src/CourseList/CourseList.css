table {
  width: 100%;
  border: 1px solid lightgrey;
  border-collapse: collapse;
  font-family: "Times New Roman", Times, serif;
}

th {
  border-bottom: 2px solid lightgrey;
  padding: 5px;
}

td {
  padding: 5px;
}

#title {
  text-align: center;
}