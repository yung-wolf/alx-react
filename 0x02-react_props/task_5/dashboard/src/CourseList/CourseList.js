// new component named CourseList. It returns a table element with an id CourseList

import React from "react";
import Proptypes from "prop-types";
import CourseListRow from "./CourseListRow";
import './CourseList.css'
import CourseShape from "./CourseShape";

const CourseList = ({ listCourses }) => {
  return (
    <table id="CourseList">
      <thead>
        {listCourses.length == 0 ?
          <CourseListRow textFirstCell="No course available yet" isHeader={true} /> :
          (
            <>
              <CourseListRow textFirstCell="Available courses" isHeader={true} />
              <CourseListRow textFirstCell="Course name" textSecondCell="Credit" isHeader={true} />
            </>
          )
        }
      </thead>

      <tbody>
        {/* <CourseListRow textFirstCell="ES6" textSecondCell="60" isHeader={false}/>
        <CourseListRow textFirstCell="Webpack" textSecondCell="20" isHeader={false}/>
        <CourseListRow textFirstCell="React" textSecondCell="40" isHeader={false} /> */}

        {/* When listCourses contains at least one element, display a row for each element showing the name of the course and the number of credits
        Make sure to add a key to help React optimize rendering */}

        {listCourses.length > 0 && listCourses.map(course => <CourseListRow key={course.id} textFirstCell={course.name} textSecondCell={course.credit} />)}

      </tbody>
    </table>
  )
}

// Define proptypes
CourseList.Proptypes = {
  listCourses: Proptypes.arrayOf(CourseShape)
};

// Define default props
CourseList.defaultProps = {
  listCourses: []
}

export default CourseList