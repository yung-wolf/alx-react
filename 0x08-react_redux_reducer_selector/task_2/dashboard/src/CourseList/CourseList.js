// new component named CourseList. It returns a table element with an id CourseList

import React from "react";
import Proptypes from "prop-types";
import CourseListRow from "./CourseListRow";
import CourseShape from "./CourseShape";
import { StyleSheet, css } from "aphrodite";

const CourseList = ({ listCourses }) => {
  return (
    <table id="CourseList" className={css(styles.table)}>
      <thead>
        {listCourses.length == 0 ?
          <CourseListRow textFirstCell="No course available yet" isHeader={true} /> :
          (
            <>
              <CourseListRow textFirstCell="Available courses" isHeader={true} />
              <CourseListRow textFirstCell="Course name" textSecondCell="Credit" isHeader={true} />
            </>
          )
        }
      </thead>

      <tbody>
        {listCourses.length > 0 && listCourses.map(course => <CourseListRow key={course.id} textFirstCell={course.name} textSecondCell={course.credit} />)}
      </tbody>
    </table>
  )
}

// hold rules for in-line styling
const styles = StyleSheet.create({
  table: {
    width: "100%",
    border: "1px solid #ccc",
    margin: "20px 0",
  },
})

// Define proptypes
CourseList.Proptypes = {
  listCourses: Proptypes.arrayOf(CourseShape)
};

// Define default props
CourseList.defaultProps = {
  listCourses: []
}

export default CourseList