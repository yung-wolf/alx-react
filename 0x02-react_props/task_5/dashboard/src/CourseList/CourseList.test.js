import React from 'react';
import { shallow } from 'enzyme';
import CourseList from './CourseList';

// Check that it renders CourseList component without crashing
it('renders without crashing', () => {
  shallow(<CourseList />);
})

// Check that it renders the 5 different rows
it('renders the 5 different rows', () => {
  const wrapper = shallow(<CourseList />);
  expect(wrapper.find('CourseListRow').length).toEqual(5);
})