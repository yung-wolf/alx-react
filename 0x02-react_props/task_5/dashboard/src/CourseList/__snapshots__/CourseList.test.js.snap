// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly if you pass an empty array 1`] = `ShallowWrapper {}`;
